### PR TITLE
Publish pasteup

### PR DESCRIPTION
**Pull request by SiAdcock**
## What does this change?

This change makes this version of pasteup publishable, for use in external libraries and applications.

## Why?

In conjunction with a published version of GUUI, this provides an alternative way to use Guardian components, rather than creating a new page or site in the `dotcom-rendering` repo.

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -119,3 +119,23 @@ flow-typed: yarn.lock
...
+	@mv packages/pasteup/*.js packages/pasteup/src
+	@mv packages/pasteup/dist/*.js packages/pasteup
+
+post-publish-pasteup:
+	$(call log, "clean up after publishing pasteup")
+	@mv packages/pasteup/src/*.js packages/pasteup
+
+publish-pasteup: clear clean-pasteup install
+	$(call log, "publishing pasteup")
+	@cd packages/pasteup && yarn pack
```
Comment (SiAdcock) [inline makefile]: Before merging, change this line to 

```
@cd packages/pasteup && yarn publish
```

File: makefile [hunk 2]
```diff
@@ -119,3 +119,23 @@ flow-typed: yarn.lock
...
 	@flow-typed install
+
+# packages #########################################
+
+clean-pasteup:
+	@rm -rf packages/pasteup/dist packages/pasteup/src
+
+pre-publish-pasteup:
+	$(call log, "building pasteup")
+	@mkdir packages/pasteup/src
```
Comment (GHaberis) [inline makefile]: Why is it necessary to create a `src` directory, copy files to that directory and use them in the build and afterwards remove that `src` directory?

File: packages/pasteup/package.json [hunk 3]
```diff
@@ -1,16 +1,23 @@
...
   "name": "@guardian/pasteup",
-  "version": "0.1.0-alpha.1",
+  "version": "1.0.0-alpha.1",
   "description": "Guardian design tokens",
-  "main": "index.js",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/guardian/dotcom-rendering"
+  },
+  "files": [
```
Comment (GHaberis) [inline packages/pasteup/package.json]: I've seen that when using `npm publish` any files listed in `.gitignore` will be excluded from the packaged tarball. We have `packages/pasteup/src` in our `.gitignore`, with `npm publish` you can add files to this `files` property in the `package.json` to ensure they're still published. Is that what's going on here? If so, do we need to specify the `src` directory in the path? eg. `src/breakpoints.js`

File: makefile (hunk 2 above)
Comment (SiAdcock) [reply makefile]: We could rename src to temp of it makes this intention clearer

Comment (GHaberis) [reply makefile]: yep I definitely think renaming it `temp` is more appropriate. `src` has other connotations in the realm of packages/libraries.

File: packages/pasteup/package.json [hunk 4]
```diff
@@ -1,16 +1,23 @@
...
+    "breakpoints.js",
+    "fonts.js",
+    "mixins.js",
+    "palette.js",
+    "icons",
+    "logos"
+  ],
   "scripts": {
-    "test": "jest"
+    "test": "jest",
```
Comment (GHaberis) [inline packages/pasteup/package.json]: What do you think about adding a yarn script to call `make clean-pasteup`? 

Currently if a developer's making changes to `pasteup` and they run `yarn postpublish` but then make a change before publishing they'd have to step back to the `dotcom-rendering` directory and run `make clean-pasteup` before they can run `yarn postpublish` again. (it'd error if they tried without cleaning up right now if `src` already exists).

Comment (SiAdcock) [reply packages/pasteup/package.json]: @ghaberis I dunno, the prepublish and postpublish hooks are not intended to be used by devs, they are automatically executed before and after yarn publish, respectively

**Pull request merged**